packages/twenty-front/src/modules/window-event/components/WindowEventProvider.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+
+import { tokenPairState } from '@/auth/states/tokenPairState';
+import { chromeExtensionIdState } from '@/client-config/states/chromeExtensionIdState';
+
+const isInFrame = () => {
+  try {
+    return window.self !== window.top;
+  } catch (e) {
+    return true;
+  }
+};
+
+export const WindowEventProvider: React.FC<React.PropsWithChildren> = ({
```

Comment:
Separate event receiving logic in an effet, and rendering logic in a component, like it's done for ClientConfig